test/OpenTelemetry.Extensions.Hosting.Tests/HostingExtensionsTests.cs

```diff
@@ -76,44 +81,78 @@ public void AddOpenTelemetryTracerProvider_HostBuilt_OpenTelemetrySdk_Registered
 
             var tracerProvider1 = host.Services.GetRequiredService<TracerProvider>();
             var tracerProvider2 = host.Services.GetRequiredService<TracerProvider>();
-
             Assert.Same(tracerProvider1, tracerProvider2);
         }
 
         [Fact]
-        public void AddOpenTelemetryTracerProvider_ServiceProviderArgument_ServicesRegistered()
+        public async Task AddOpenTelemetryTracingDetectMultipleCalls()
         {
             var testInstrumentation = new TestInstrumentation();
+            int instrumentationCallBackCount = 0;
 
-            var services = new ServiceCollection();
-            services.AddSingleton(testInstrumentation);
-            services.AddOpenTelemetryTracing((provider, builder) =>
+            var builder = new HostBuilder().ConfigureServices(services =>
             {
-                builder.AddInstrumentation(() => provider.GetRequiredService<TestInstrumentation>());
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+                services.AddOpenTelemetryTracing();
+                services.Configure<TracerProviderBuilder>((builder) =>
```

Comment:
Would it be possible to use `services.AddOpenTelemetryTracing((builder) => { ... })` even for extensions rather then `services.Configure<TracerProviderBuilder>((builder) => {...})`. I presume the tests are to demonstrate how a consumer of the library would use it. And if that's so, using `services.Configure<T>` to mutate TracerProviderBuilder seems like we are leaking implementation detail. (We can still use the same mechanics behind the scenes).